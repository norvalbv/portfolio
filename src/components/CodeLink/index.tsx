import React, { ReactElement } from 'react';
import { CodeIcon } from 'components/SVG';
import { classNames } from 'utils';

export type CodeLinkProps = {
  requireIcon?: boolean;
  className?: string;
  link: string;
};

const CodeLink = ({ className, link, requireIcon = true }: CodeLinkProps): ReactElement => {
  return (
    <div className={classNames('group', className)}>
      <a
        href={link}
        target="_blank"
        rel="noopener noreferrer"
        className="items-center justify-center gap-2 lg:flex"
      >
        <p className="text-xs opacity-100 transition-opacity duration-300 lg:opacity-0 lg:group-hover:opacity-100">
          View Code
        </p>
        {requireIcon && <CodeIcon className="hidden cursor-pointer lg:block" />}
      </a>
    </div>
  );
};

export default CodeLink;
