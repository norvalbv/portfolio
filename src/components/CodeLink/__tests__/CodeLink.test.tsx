import React, { FC } from 'react';
import renderer from 'react-test-renderer';

import CodeLink, { CodeLinkProps } from '..';

const CodeLinkWithAllProps: FC<Required<CodeLinkProps>> = CodeLink;

describe('<CodeLink />', () => {
  test('should render correctly with mandatory props', () => {
    const tree = renderer.create(<CodeLink link="/" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render correctly with all props', () => {
    const tree = renderer
      .create(<CodeLinkWithAllProps requireIcon={false} className="bg-red-500" link="/" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
