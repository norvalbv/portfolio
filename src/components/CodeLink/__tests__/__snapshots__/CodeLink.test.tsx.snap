// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeLink /> should render correctly with all props 1`] = `
<div
  className="group bg-red-500"
>
  <a
    className="items-center justify-center gap-2 lg:flex"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <p
      className="text-xs opacity-100 transition-opacity duration-300 lg:opacity-0 lg:group-hover:opacity-100"
    >
      View Code
    </p>
  </a>
</div>
`;

exports[`<CodeLink /> should render correctly with mandatory props 1`] = `
<div
  className="group"
>
  <a
    className="items-center justify-center gap-2 lg:flex"
    href="/"
    rel="noopener noreferrer"
    target="_blank"
  >
    <p
      className="text-xs opacity-100 transition-opacity duration-300 lg:opacity-0 lg:group-hover:opacity-100"
    >
      View Code
    </p>
    <svg
      className="hidden cursor-pointer lg:block"
      fill="none"
      height={20}
      stroke="currentColor"
      strokeWidth={1.5}
      viewBox="0 0 24 24"
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M17.25 6.75L22.5 12l-5.25 5.25m-10.5 0L1.5 12l5.25-5.25m7.5-3l-4.5 16.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  </a>
</div>
`;
